src/Integration.Vsix/SolutionAnalysisRequester.cs

```diff
@@ -0,0 +1,106 @@
+        private Option<bool> option;
+        private MethodInfo optionServiceSetOptionsMethod;
+        private MethodInfo optionServiceGetOptionsMethod;
+
+        public SolutionAnalysisRequester(Workspace workspace)
+        {
+            if (workspace == null)
+            {
+                throw new ArgumentNullException(nameof(workspace));
+            }
+
+            this.workspace = workspace;
+            this.FindOptionService();
+            this.FindFullSolutionAnalysisOption();
+        }
+
+        private void FindOptionService()
+        {
+            var types = typeof(PerLanguageOption<int>).Assembly.GetTypes();
+            var interfaces = types.Where(t => t.IsInterface);
+
+            var optionServiceType = interfaces.FirstOrDefault(t => t.Name == "IOptionService");
+            if (optionServiceType == null)
+            {
+                return;
+            }
+
+            this.optionService = this.workspace.Services.GetType()
+                .GetMethod("GetService")
+                ?.MakeGenericMethod(optionServiceType)
+                ?.Invoke(workspace.Services, null);
+
+            this.optionServiceSetOptionsMethod = optionServiceType?.GetMethod("SetOptions");
+            this.optionServiceGetOptionsMethod = optionServiceType?.GetMethod("GetOptions");
+        }
+
+        private void FindFullSolutionAnalysisOption()
+        {
+            //there's no public type in the DLL, so we try finding it next to Microsoft.CodeAnalysis
+            var path = Path.Combine(
+                new FileInfo(typeof(Accessibility).Assembly.Location).Directory.FullName,
+                "Microsoft.CodeAnalysis.Features.dll");
+
+            if (!File.Exists(path))
+            {
+                return;
+            }
+
+            this.option = (Option<bool>)Assembly.LoadFile(path)
+                .GetTypes()
+                .Where(t => t.IsClass && t.IsSealed && t.IsAbstract)
+                .FirstOrDefault(t => t.Name == "RuntimeOptions")
+                ?.GetField("FullSolutionAnalysis")
+                ?.GetValue(null);
+        }
+
+        private void FlipFullSolutionAnalysisFlag()
+        {
+            if (this.optionService == null ||
+                this.option == null)
+            {
+                return;
+            }
+
+            var options = this.optionServiceGetOptionsMethod?.Invoke(optionService, null) as OptionSet;
+            if (options == null)
+            {
+                return;
+            }
+
+            var optionValue = options.GetOption(this.option);
+            var newOptions = options.WithChangedOption(this.option, !optionValue);
+            this.optionServiceSetOptionsMethod?.Invoke(this.optionService, new object[] { newOptions });
+        }
+
+        /// <summary>
+        /// Forces a full solution reanalysis
+        /// </summary>
+        public void ReanalyzeSolution()
+        {
+            this.FlipFullSolutionAnalysisFlag();
```

Comment:
Add a comment stating that these repeated calls to `FlipFullSolutionAnalysisFlag` are intentional, and not something like the result of a bad merge.